helm/azure-operator-chart/templates/deployment.yaml

```diff
@@ -21,37 +21,37 @@ spec:
       annotations:
         releasetime: {{ $.Release.Time }}
       labels:
-        app: azure-operator
+        app: {{ tpl .Values.resource.default.name . }}
         version: {{ .Chart.Version }}
     spec:
       volumes:
-      - name: azure-operator-configmap
+      - name: {{ .Values.project.name }}-configmap
```

Comment:
I had the same question when flattening: why do we sometimes use `.Values.project.name` and some other times `tpl .Values.resource.default.name`?

Reply:
I asked about this this in the `aws-operator` PR and was thinking the reason is probably that you want to have multiple operators running in parallel so resource names are distinct per release/version, but the YAML content for manifests are still the same across different versions.

Reply:
Stuff in `values.yaml` is not templated. So if you used `{{ .Values.resource.default.name . }}` the output would be `{{ .Release.Name | replace "." "-" | trunc 47 }}`. with `tpl` it actually templates the value output before rendering.

Reply:
I misinterpreted the original question. I talked with @fiunchinho and the outcome is we shouldn't use `.Values.project.name` in general.